front-end/src/components/analysis-page/IntentMenu.jsx

```diff
@@ -41,6 +44,14 @@ class IntentMenu extends Component {
         this.setState({inputValue: event.target.value});
     }
 
+    handleStarChange() {
```

Comment:
super pedantic but I think we may benefit from a less ambiguous function name here because this is similar to what we call the action of toggling star/flag on the actual intents. Maybe sneak the word filter into here?